m3/src/main/java/com/uber/m3/tally/m3/M3Reporter.java

```diff
@@ -137,51 +126,46 @@ private M3Reporter(Builder builder) {
 
             client = new M3.Client(protocolFactory.getProtocol(transport));
 
-            calcProtocol = new TCompactProtocol.Factory().getProtocol(new TCalcTransport());
-            calc = (TCalcTransport) calcProtocol.getTransport();
+            payloadCapacity = calculatePayloadCapacity(builder.maxPacketSizeBytes, builder.metricTagSet);
 
-            freeBytes = calculateFreeBytes(builder.maxPacketSizeBytes, builder.metricTagSet);
-
-            maxProcessorWaitUntilFlushMillis = builder.maxProcessorWaitUntilFlushMillis;
+            maxBufferingDelay = Duration.ofMillis(builder.maxProcessorWaitUntilFlushMillis);
 
             bucketIdTagName = builder.histogramBucketIdName;
             bucketTagName = builder.histogramBucketName;
             bucketValFmt = String.format("%%.%df", builder.histogramBucketTagPrecision);
 
             metricQueue = new LinkedBlockingQueue<>(builder.maxQueueSize);
 
-            executor = builder.executor;
+            executor = builder.executor != null ? builder.executor : Executors.newFixedThreadPool(NUM_PROCESSORS);
```

Comment:
nit: can you move this to the builder

Reply:
Specifically moved it out, b/c builder wasn't shutting down the executor when it was overridden

Reply:
could you only construct in the builder when required? that way the shutdown is guaranteed because it's done here. and we don't construct a useless one